src/parameterized_tendencies/microphysics/microphysics_wrappers.jl

```diff
@@ -324,3 +325,182 @@ function compute_precipitation_sinks!(
     @. Sqₛᵖ += Sᵖ
     #! format: on
 end
+
+#####
+##### 2M microphysics
+#####
+
+"""
+    aerosol_activation_sources(cm_params, thp, ts, dt)
+
+ - cm_params - CloudMicrophysics parameters struct for cloud water or ice condensate
+ - thp - Thermodynamics parameters struct
+ - ts - thermodynamics state
+ - qₚ - precipitation (rain or snow) specific humidity
+ - n_dp - number concentration droplets (liquid or ice) per mass
+ _ n_dp_prescribed - prescribed number concentration of droplets (liquid or ice) per mass
+ - dt - model time step
+
+Returns the activation rate. #TODO This function temporarily computes activation rate 
+based on mass rates and a prescribed droplet mass (no activation parameterization yet).
+"""
+function aerosol_activation_sources(
+    cm_params::Union{CMP.CloudLiquid{FT}, CMP.CloudIce{FT}},
+    thp,
+    ts,
+    qₚ,
+    n_dp,
+    n_dp_prescribed,
+    dt,
+) where {FT}
+    r_dp = FT(2e-6) # 2 μm
+    m_dp = 4 / 3 * π * r_dp^3 * 1000 # ρ_water = 1000 kg/m^3
+    Sn = cloud_sources(cm_params, thp, ts, qₚ, dt) / m_dp
+
+    return ifelse(
+        Sn > FT(0),
+        triangle_inequality_limiter(Sn, limit((n_dp_prescribed - n_dp), dt, 2)),
+        -triangle_inequality_limiter(abs(Sn), limit(n_dp, dt, 2)),
+    )
+end
+
+"""
+    compute_warm_precipitation_sources_2M!(Sᵖ, S₂ᵖ, Snₗᵖ, Snᵣᵖ, Sqₗᵖ, Sqᵣᵖ, ρ, nₗ, nᵣ, qₗ, qᵣ, dt, sb, thp)
+
+ - Sᵖ, S₂ᵖ - temporary containters to help compute precipitation source terms
+ - Snₗᵖ, Snᵣᵖ, Sqₗᵖ, Sqᵣᵖ - cached storage for precipitation source terms
+ - ρ - air density
+ - nₗ, nᵣ - cloud liquid and rain number concentration per mass
+ - qₗ, qᵣ - cloud liquid and rain specific humidity
+ - ts - thermodynamic state (see td package for details)
+ - dt - model time step
+ - thp, mp - structs with thermodynamic and microphysics parameters
+
+Computes precipitation number and mass sources due to warm precipitation processes based on the 2-moment 
+Seifert and Beheng (2006) scheme.
+"""
+function compute_warm_precipitation_sources_2M!(
```

Comment:
overall in this function, it would be useful if you added some comments to every source that explains / accounts for the negative and/or double-negative signs that are sometimes inserted.